### ISSUE TITLE
[regression] 0.24.1-1 breaks multi-line prompt resize 

### ISSUE THREAD

**Issue by zbirenbaum** [kovidgoyal/kitty#4495]
Upgrading kitty to .24.1-1 causes multiline prompts to leave junk in random areas of the buffer similar to versions of kitty prior to (i believe) some changes to how cursor and text reflow is tracked. I know, because while not this bad, versions prior to .24.0-1 had similar issues. The issue is completely fixed by downgrading to 0.24.0-1, so I am not sure if this is caused by the new shell-integration package or the .24.1-1 changes. 

For reference I am using powerlevel10k and am on arch linux.

**Comment by kovidgoyal** [comment]
You need to enable kitty shell integration in your powerlevel10k config.
Read the powerlevel10k readme about prompt resizing.

**Comment by kovidgoyal** [comment]
And since the only relevant changes in 0.24.1 were by the author of powerlevel10k, ping @romkatv

**Comment by zbirenbaum** [comment]
Apologies I was not aware that p10k was directly involved here. I did some deep reading on p10k resizing a while ago but those issues were from a couple years back and with zsh itself. Thank you for the guidance, I'll take a look there.

I didn't see any direct references to this in the change log but it may save some others from making a post like this one if there's a notice about this specific issue. I was really excited when it was finally fixed last patch so I may have jumped the gun on seeing it break again.
